docs/src/extend/custom-parsers.md

```diff
@@ -8,63 +8,69 @@ eleventyNavigation:
+ESLint custom parsers let you extend ESLint to support linting new non-standard JavaScript language features or custom syntax in your code. A parser is responsible for taking your code and transforming it into an abstract syntax tree (AST) that ESLint can then analyze and lint.
 
-* `ast` should contain the AST.
-* `services` can contain any parser-dependent services (such as type checkers for nodes). The value of the `services` property is available to rules as `context.parserServices`. Default is an empty object.
-* `scopeManager` can be a [ScopeManager](./scope-manager-interface) object. Custom parsers can use customized scope analysis for experimental/enhancement syntaxes. Default is the `ScopeManager` object which is created by [eslint-scope](https://github.com/eslint/eslint-scope).
-    * Support for `scopeManager` was added in ESLint v4.14.0. ESLint versions which support `scopeManager` will provide an `eslintScopeManager: true` property in `parserOptions`, which can be used for feature detection.
-* `visitorKeys` can be an object to customize AST traversal. The keys of the object are the type of AST nodes. Each value is an array of the property names which should be traversed. Default is [KEYS of `eslint-visitor-keys`](https://github.com/eslint/eslint-visitor-keys#evkkeys).
-    * Support for `visitorKeys` was added in ESLint v4.14.0. ESLint versions which support `visitorKeys` will provide an `eslintVisitorKeys: true` property in `parserOptions`, which can be used for feature detection.
+## Creating a Custom Parser
 
-You can find an ESLint parser project [here](https://github.com/typescript-eslint/typescript-eslint).
+A custom parser is a JavaScript object with either a `parse` or `parseForESLint` method. The `parse` method only returns the AST, whereas `parseForESLint` also returns additional values that let you customize the behavior of the parser even more.
 
-```json
-{
-    "parser": "./path/to/awesome-custom-parser.js"
-}
-```
+Both methods should take in the source code as the first argument, and an optional configuration object as the second argument, which is provided as [`parserOptions`](../use/configure/language-options#specifying-parser-options) in a configuration file.
 
 ```javascript
-var espree = require("espree");
-// awesome-custom-parser.js
-exports.parseForESLint = function(code, options) {
-    return {
-        ast: espree.parse(code, options),
-        services: {
-            foo: function() {
-                console.log("foo");
-            }
-        },
-        scopeManager: null,
-        visitorKeys: null
-    };
+// customParser.js
+
+const espree = require("espree");
+
+// Logs the time before and after parsing each file.
```

Comment:
Actually, only the duration of parsing each file is logged, i.e. only `console.timeEnd()` produces an output. I'm sorry if this was not clear from my last comment.

Reply:
```suggestion
// Logs the duration it takes to parse each file.
```